#8336 - triggers description (#7)
G1ANT-Robot/G1ANT.Addon.Core: G1ANT.Addon.Core/Triggers/MailTrigger.md

> # MailTrigger

Author: ```suggestion
# Mail Trigger
```

> MailTrigger provides information about incoming emails.

Author: ```suggestion
Mail Trigger provides information about incoming emails.
```

## Initial Arguments

| Trigger Argument | Required | Default value | Description |
| -------- | ---- | -------- | ------------- |
| `Host` | yes |  | IMAP server address |
| `Login` | yes |  | Login (e.g.: email address) for the account the emails will be sent from |
| `Password` | yes |  | Password for this email account |
| `Frequency` | no | 30000 | Time interval between updates on incoming emails (in milliseconds) |
| `Port` | yes |  | Incoming mail server port |

## Task Arguments

Arguments that are passed to the script on its execution for each incoming email:

| Trigger Argument | Description |
| -------- | ---- |
| `Title` | Title of an incoming mail |
| `AttachmentNames` | List of attachments in an incoming mail |
| `Content` | Incoming mail content (text) |
| `From` | Sender's email address |
| `To` | Recipient's email address |
| `HtmlBody` | Incoming mail content (HTML) |
| `Uid` | Unique identifier of a mail given by a server |
| `Date` | Sent date |
| `Priority` | Priority set by a sender |

> ## Example of Defining a MailTrigger in Settings

Author: ```suggestion
## Example of Defining a Mail Trigger in Settings
```

![MailTrigger in Settings](https://manual.g1ant.com/link/G1ANT.Manual/-assets/mailtriggerexample.png)

> ## Example of Defining a MailTrigger in the Config File

Author: ```suggestion
## Example of Defining a Mail Trigger in the Config File
```

## Example

> The script executed by the MailTrigger can make use of task arguments mentioned earlier. The following line included in the script will display the title and the recipient’s name of an email, which triggered the script execution:

Author: ```suggestion
The script executed by the Mail Trigger can make use of task arguments mentioned earlier. The following line included in the script will display the title and the recipient’s name of an email, which triggered the script execution:
```
